### PR TITLE
Add free-trial page

### PR DESCRIPTION
**Pull request by dtuite**
Replace the typeform flow with a form directly on the website.

There are a few chunks of work involved in this:

1. The addition of new form components, checkbox, radio, TextField etc. so I can make a custom styled form.
2. The addition of a `/free-trial/` page, which uses these components to ask the user some questions, and send the results to Netlify forms.
3. The addition of a PageMargins component because there was some padding baked into the root elements of the site that I didn't want to exist on the free trial page. They can be selectively added back in by using this component.

Direct link to change: https://deploy-preview-359--roadie.netlify.app/free-trial/

[ch3811]

### CODE REVIEW

File: src/components/actions/EmailCaptureForm.js [hunk 1]
```diff
@@ -1,71 +1,30 @@
...
 
 const styles = (theme) => ({
   inputWrapper: {
     width: '100%',
     display: 'flex',
-    marginBottom: 8,
     flexDirection: 'column',
   },
 
-  input: {
```
Comment (dtuite) [inline src/components/actions/EmailCaptureForm.js:17]: This stuff all lives in the `TextField` component now

File: src/components/layouts/PageMargins.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    paddingRight: 16,
+  },
+});
+
+const PageMargins = ({ children }) => {
+  const classes = useStyles();
+  return <div className={classes.root}>{children}</div>;
+};
+
+export default PageMargins;
```
Comment (dtuite) [inline src/components/layouts/PageMargins.js:16]: I wanted a background color on half of the free trial page, but there was left & right padding built into the `LayoutControl` component which is used on basically every page of the site. I split out this padding functionality into this new component which can be optionally used on a case by case basis.

File: content/docs/getting-started/getting-started-for-admins/index.md [hunk 3]
```diff
@@ -88,4 +88,4 @@ The GitHub permissions that Roadie needs are all read only. You can also choose
 
 Let's [add a component to Backstage](/docs/getting-started/adding-components/).
 
-Also you can [add add another user to Backstage](/docs/getting-started/user-management/).
\ No newline at end of file
+Also you can [add add another user to Backstage](/docs/getting-started/user-management/).
```
Comment (Xantier) [inline content/docs/getting-started/getting-started-for-admins/index.md:91]:
```suggestion
Also you can [add another user to Backstage](/docs/getting-started/user-management/).
```

File: src/components/CallToAction/ExtendedGetInstanceCallToAction.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+const SCM_TOOLS = [{
+  value: 'github-enterprise-cloud',
+  label: 'GitHub Enterprise (Cloud)',
+}, {
+  value: 'github-enterprise-on-prem',
+  label: 'GitHub Enterprise (On-prem)',
+}, {
+  value: 'github',
+  label: 'GitHub (Not sure)',
```
Comment (Xantier) [inline src/components/CallToAction/ExtendedGetInstanceCallToAction.js:26]: Maybe 'standard' or something along those lines instead of 'Not sure'?

**Review by Xantier — DISMISSED** [review]

File: src/components/CallToAction/ExtendedGetInstanceCallToAction.js (hunk 4 above)
Comment (dtuite) [reply src/components/CallToAction/ExtendedGetInstanceCallToAction.js:26]: @Xantier I was actually intending that option for people who think "I know we use GitHub but I'm not sure if it's the cloud version or the on-prem version". Thinking about it again, though... I have left out non-enterprise versions of GitHub, and I don't have a "Not sure" version of GitLab or Bitbucket.

Perhaps the best layout would instead be...

```js
const SCM_TOOLS = [{
  value: 'github-cloud',
  label: 'GitHub Cloud',
}, {
  value: 'github-on-prem',
  label: 'GitHub On-prem',
}, {
  value: 'gitlab-cloud',
  label: 'Gitlab Cloud',
}, {
  value: 'gitlab-on-prem',
  label: 'Gitlab On-prem',
}, {
  value: 'bitbucket-cloud',
  label: 'Bitbucket Cloud',
}, {
  value: 'bitbucket-server',
  label: 'Bitbucket Server',
}, {
  value: 'other',
  label: 'Other',
}];
```

Comment (Xantier) [reply src/components/CallToAction/ExtendedGetInstanceCallToAction.js:26]: :+1: That makes sense to me

**Review by Xantier — APPROVED** [review]

File: src/components/CallToAction/ExtendedGetInstanceCallToAction.js (hunk 4 above)
Comment (dtuite) [reply src/components/CallToAction/ExtendedGetInstanceCallToAction.js]: I pushed a commit to switch to the setup described above.

**Review by Xantier — APPROVED** [review]

**Pull request merged**